bugmon_tc/monitor/tasks.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+
+from taskcluster import slugId
+
+
+class ProcessorTask(object):
+    """
+    Helper class for generating processor tasks
+    """
+
+    def __init__(self, parent_id, src, dest=None, deps=None):
+        """
+
+        :param parent_id: ID of parent task
+        :param src: Path to monitor artifact
+        :param dest: Path to artifact generated by task
+        :param deps: Additional dependencies needed by the task
+        """
+        self.id = slugId()
+        self.parent_id = parent_id
+        self.src = src
+        self.dest = dest
+
+        self.deps = [parent_id]
+        if deps is not None:
+            self.deps.extend(deps)
+        self.scope = [f"queue:get-artifact:project/fuzzing/bugmon/{self.src}"]
+        self.worker = "bugmon-processor"
+
+    @property
+    def env(self):
+        """ Environment variables for the task """
+        return {
+            "BUGMON_ACTION": "PROCESS",
+            "MONITOR_ARTIFACT": self.src,
+            "PROCESSOR_ARTIFACT": self.dest,
+        }
+
+    @property
+    def task(self):
+        """ Task definition """
+        return {
+            "taskGroupId": self.parent_id,
+            "dependencies": self.deps,
+            "deadline": {"$fromNow": "1 hour"},
+            "expires": {"$fromNow": "1 week"},
+            "provisionerId": "proj-fuzzing",
+            "metadata": {
+                "description": "",
+                "name": "Bugmon hook",
+                "owner": "fuzzing+taskcluster@mozilla.com",
+                "source": "https://github.com/MozillaSecurity/bugmon",
+            },
+            "payload": {
+                "artifacts": {
+                    # Path for artifact generated by task
+                    "project/fuzzing/bugmon": {
```

Comment:
If you change the `type` to `directory` and remove `self.dest` from `path`, then you don't need to worry about whether `dest` is set or not. Otherwise you're missing `dest` from the artifact key.

Reply:
Done